src/sentry/digests/utilities.py

```diff
@@ -23,3 +24,134 @@ def get_digest_metadata(digest):
                     end = record.datetime
 
     return start, end, counts
+
+
+def get_personalized_digests(project_id, digest, user_ids):
+    """
+    get_personalized_digests(project: Project, digest: Digest, users: Set[User]) -> Iterator[User, Digest]
+    """
+    # TODO(LB): I Know this is inefficent.
+    # In the case that ProjectOwnership does exist, I do the same query twice.
+    # Once with this statement and again with the call to ProjectOwnership.get_actors()
+    # Will follow up with another PR to reduce the number of queries.
+    if ProjectOwnership.objects.filter(project_id=project_id).exists():
+        events = get_events_from_digest(digest)
+        events_by_actor = build_events_by_actor(project_id, events)
+        events_by_user = convert_actors_to_user_set(events_by_actor, user_ids)
+        for user_id in user_ids:
+            yield user_id, build_custom_digest(digest, events_by_user[user_id])
```

Comment:
Is it possible for the `user_id` to not present in the `events_by_user` mapping?

Reply:
Yes it is. I'm a bit surprised my tests missed that case. I thought I had tested that. I'll double check and if not add one. But you're correct.